### PR TITLE
Moved to primitive longs as primary snowflake storage type

### PR DESCRIPTION
**Pull request by MinnDevelopment**
In this experiment I changed almost all `id` fields to use `long` over `String`.
To increase performance and memory usage we use `trove4j` which provides HashMap implementations that use backing primitive arrays instead of `Long` arrays.

Test:
- [x] Compiles with no errors
- [x] Runs with no JSONExceptions
- [x] Runs with large sharded bots (LewdBot, DabBot, etc.)
- [x] Benchmarking

### CODE REVIEW

**Comment by MinnDevelopment** [comment]
First benchmarking checks (VisualVM) have shown a major decrease in memory usage for this experiment.
Specific data is not yet retrieved.

**Comment by DV8FromTheWorld** [comment]
Would like some graphs / statistics posted here of a comparison between `experimental/cache` and `master`'s 3.0 release.

**Comment by DV8FromTheWorld** [comment]
The graphs that were posted in Discord, move them to here for public comparisons.
Will review the code in the coming days.

**Comment by MinnDevelopment** [comment]
Here are the comparison graphs.
image
> Taken using Java VisualVM

File: src/main/java/net/dv8tion/jda/client/entities/impl/ApplicationImpl.java [hunk 1]
```diff
@@ -62,7 +63,7 @@ public ApplicationImpl(final JDA api, final JSONObject object)
         if (this.hasBot())
             return new RestAction.EmptyRestAction<>(this.bot);
 
-        return new RestAction<Application.Bot>(this.api, Route.Applications.CREATE_BOT.compile(this.id), null)
+        return new RestAction<Application.Bot>(this.api, Route.Applications.CREATE_BOT.compile(getId()), null)
```
Comment (Almighty-Alpaca) [inline src/main/java/net/dv8tion/jda/client/entities/impl/ApplicationImpl.java]: why don't you use the long for Routes? would need one string creation less per RestAction

Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/client/entities/impl/ApplicationImpl.java]: The compile method only accepts String but maybe I can change that

**Comment by DV8FromTheWorld** [comment]
This is on hold until #287 and #302 are merged.

**Comment by MinnDevelopment** [comment]
This will be updated to fix conflicts once #302 is merged

**Comment by MinnDevelopment** [comment]
@DV8FromTheWorld This is now ready for review.

**Pull request merged**